src/components/free-willy/MyBids.tsx

```diff
@@ -25,54 +26,160 @@ const columns = [
+            }
+            if(bluna_bids.length > 0) {
+                setTransactionData([activateBids(network.contracts.bluna)]);
+            }
+        }
+    }
+
+    const retract = () => {
+        setRetracting(true)
+        const bids_to_retract: MsgExecuteContract[] = []
+        selectionModel?.forEach(row => {
+            bids_to_retract.push(retractBid(row.id));
+        });
+        setTransactionData(bids_to_retract)
+    }
+
+    const success = () => {
+        if(onClaim) {
+            onClaim();
+        }
+    }
 
     return (
         <Stack sx={{padding: '10px'}}>
-            <Typography variant="h4" sx={{margin: '10px'}}>
-                My Bids
-            </Typography>
+            <Grid container alignItems="center">
+                <Grid item md={8} xs={6}>
+                    <Typography variant="h4" sx={{margin: '10px'}}>
+                            My Bids
+                    </Typography>
+                </Grid>
+                <Grid item md={4} xs={6} alignItems="right">
+                    <Button sx={{margin:"5px"}} variant="contained" disabled={!retractable} onClick={retract}>Retract Bid{selectionModel?.length!=1?"s":""}</Button>
+                    <Button sx={{margin:"5px"}} variant="contained" disabled={!activatable} onClick={activate}>Activate Bid{selectionModel?.length!=1?"s":""}</Button>
+                </Grid>
+            </Grid>
+
             <div style={{ height: 300 }}>
                 <DataGrid
+                    sx={{
+                        '&.MuiDataGrid-root .MuiDataGrid-cell:focus': {
+                            outline: 'none',
+                        },
+                    }}
                     rows={rows}
                     columns={columns}
                     pageSize={30}
                     rowsPerPageOptions={[30]}
                     checkboxSelection
-                    disableSelectionOnClick
+                    disableColumnSelector
+                    disableDensitySelector
+                    onSelectionModelChange={(ids) => {
+                        const selectedIDs = new Set(ids);
+                        const selectedRowData = rows.filter((row) => selectedIDs.has(row.id.toString()));
+                        setSelectionModel(selectedRowData);
+                        if(selectedRowData.length>0) {
+                            setRetractable(true)
+                            var a = true;
+                            selectedRowData.forEach(row => {
+                                if (row.bid_status != "Ready for activation") {
+                                    a = false;
+                                }
+                            });
+                            setActivatable(a)
+                        } else {
+                            setRetractable(false)
+                            setActivatable(false)
+                        }
+                        console.log(activatable)
```

Comment:
remove console logs.  Also consider defining this function above out of the JSX logic.